pfrl/agents/goal_conditioned_td3.py

```diff
@@ -19,6 +20,29 @@ def default_target_policy_smoothing_func(batch_action):
     return torch.clamp(batch_action + noise, -1, 1)
 
 
+def _mean_or_nan(xs):
+    """Return its mean a non-empty sequence, numpy.nan for a empty one."""
+    return np.mean(xs) if xs else np.nan
+
+
+class TemperatureHolder(nn.Module):
+    """Module that holds a temperature as a learnable value.
+
+    Args:
+        initial_log_temperature (float): Initial value of log(temperature).
+    """
+
+    def __init__(self, initial_log_temperature=0):
+        super().__init__()
+        self.log_temperature = nn.Parameter(
+            torch.tensor(initial_log_temperature, dtype=torch.float32)
+        )
+
+    def forward(self):
+        """Return a temperature as a torch.Tensor."""
+        return torch.exp(self.log_temperature)
+
+
```

Comment:
remove if not necessary

Reply:
it's to experiment with some things later on